src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -164,14 +206,38 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 			args.request.values.belongs_to__application != null &&
 			affectedIds.length !== 0
 		) {
-			await args.api.delete({
-				resource: 'service_install',
-				options: {
-					$filter: {
-						device: { $in: affectedIds },
+			const deleteHandler = (
+				api: sbvrUtils.PinejsClient,
+				deviceIds: number[],
+				serviceIds: number[],
+			) =>
+				api.delete({
+					resource: 'service_install',
+					options: {
+						$filter: {
+							device: { $in: deviceIds },
+							installs__service: { $in: serviceIds },
+						},
 					},
-				},
-			});
+				});
+
+			await actOnAppServiceInstalls(
+				args.api,
+				args.request.values.belongs_to__application,
+				affectedIds,
+				deleteHandler,
+			);
```

Comment:
I'd rather see this done in a single request as it's acting on existing service installs, and avoid the extra work that is only required in the case we need to find missing service installs
```ts
				api.delete({
					resource: 'service_install',
					options: {
						$filter: {
							device: { $in: deviceIds },
							installs__service: {
								$any: {
									$alias: 's',
									$expr: {
										s: { 
											application: {
												$any: {
													$alias: 'a',
													$expr: { a: { id: appId } },
												},
											},
										},
									},
								},
							},
						},
					},
				});

Reply:
I re-wrote the code to create them directly without any comparison with existing service installs, since in this case there won't be any.

Reply:
@toochevere I was talking about deleting the service installs in a single request, I'm not sure where creating service installs came from?

Reply:
https://github.com/balena-io/open-balena-api/blob/master/src/features/ci-cd/hooks/service-installs.ts#L167

The original code, on moving apps, deleted _all_ the previous installs and created the new ones for the target app.  We are now deleting _only_ the service installs for the previous app, and then as before creating the new ones for the target app.

Reply:
I understand that behavior but the response to my initial message is what has confused me - I proposed deleting the existing service installs for the current app in a single api query instead of many, and then continue with the creation as normal

Reply:
oh, sorry.  I misunderstood. 🤦🏻   In any case, the creation is now also more optimized.  I'll make the deletion change

Reply:
done

Reply:
replaced the deletion code with the snippet you gave me.